projects/postgresql/fuzzer/simple_query_fuzzer.c

```diff
@@ -108,10 +109,12 @@ int __attribute__((constructor)) Initialize(void) {
   int argc = 4;
   char *argv[4];
   argv[0] = "tmp_install/usr/local/pgsql/bin/postgres";
-  argv[1] = "-D\"temp/data\"";
+  argv[1] = "-D\"/tmp/query_db/data\"";
   argv[2] = "-F";
   argv[3] = "-k\"/tmp/pg_dbfuzz\"";
-	
+
+  system("tar -xvf query_db.tar.gz -C /tmp/");
```

Comment:
if you want to extract into a subdir, you have to mkdir it first, but that can be done in a single system() call: system("mkdir /tmp/query_db && tar ... -C /tmp/query_db");